### ISSUE TITLE
Data is not valid - Field ID exceeds maximum length of

### ISSUE THREAD

**Issue by Nachtlichtermeer** [nilsteampassnet/TeamPass#3729]
### Steps to reproduce
1. edit some item
2. klick 'save'

### Expected behaviour
changes should be saved

### Actual behaviour
Error is shown: 'Data is not valid - Field ID exceeds maximum length of'

### Server configuration
**Operating system**:
Ubuntu 22.04
**Web server:**
Apache
**Database:**
MySQL
**PHP version:**
8.1
**Teampass version:** 
3.0.9 commit 643a067cc747ca8de23c52d44b023bcb95bf3fc5
**Teampass configuration file:**

**Updated from an older Teampass or fresh install:**
PLEASE attach to this issue the file `/includes/config/tp.config.php`.

### Client configuration
**Browser:**
Firefox 113.0.1
**Operating system:**
Ubuntu 22.04
#### Web server error log
nothing
Insert your webserver log here
`foo - - [23/May/2023:09:36:06 +0200] "POST /sources/items.queries.php HTTP/1.1" 200 971 "https://foo/index.php?page=items" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/113.0"
`

#### Log from the web-browser developer console (CTRL + SHIFT + i)
nothing
Insert the log here and especially the answer of the query that failed.
```

**Comment by Nachtlichtermeer** [comment]
latest working commit is ff9b527a4ebd681085a0361a163f7b11d877eb12

**Comment by Nachtlichtermeer** [comment]
if i comment out the following code in sources/items.queries.php it works:
```
                $dataCheck = validateDataFields(prefixTable('items'), $dataReceived);
                if ($dataCheck['state'] !== true) {
                    echo (string) prepareExchangedData(
                        $SETTINGS['cpassman_dir'],
                        array(
                            'error' => true,
                            'message' => langHdl('error_data_not_valid').' - '.$dataCheck['message'],
                        ),
                        'encode'
                    );
                    break;
                }
```
there is something wrong with this check

**Comment by nilsteampassnet** [comment]
@Nachtlichtermeer Can you please run next query
```
SELECT AUTO_INCREMENT FROM information_schema.TABLES WHERE TABLE_SCHEMA = "teampass" AND TABLE_NAME = "teampass_items";
```

**Comment by Nachtlichtermeer** [comment]
`mysql> SELECT AUTO_INCREMENT FROM information_schema.TABLES WHERE TABLE_SCHEMA = "teampass" AND TABLE_NAME = "teampass_items";
+----------------+
| AUTO_INCREMENT |
+----------------+
|           2519 |
+----------------+
1 row in set (0,00 sec)`

**Comment by Nachtlichtermeer** [comment]
```
mysql> DESCRIBE `teampass_items`;
+---------------------------+--------------+------+-----+---------+----------------+
| Field                     | Type         | Null | Key | Default | Extra          |
+---------------------------+--------------+------+-----+---------+----------------+
| id                        | int          | NO   | PRI | NULL    | auto_increment |
| label                     | varchar(500) | NO   |     | NULL    |                |
| description               | text         | YES  |     | NULL    |                |
| pw                        | text         | YES  |     | NULL    |                |
| pw_iv                     | text         | YES  |     | NULL    |                |
| pw_len                    | int          | NO   |     | 0       |                |
| url                       | varchar(500) | YES  |     | NULL    |                |
| id_tree                   | varchar(10)  | YES  |     | NULL    |                |
| perso                     | tinyint(1)   | NO   |     | 0       |                |
| login                     | varchar(200) | YES  |     | NULL    |                |
| inactif                   | tinyint(1)   | NO   |     | 0       |                |
| restricted_to             | varchar(200) | YES  | MUL | NULL    |                |
| anyone_can_modify         | tinyint(1)   | NO   |     | 0       |                |
| email                     | varchar(100) | YES  |     | NULL    |                |
| notification              | varchar(250) | YES  |     | NULL    |                |
| viewed_no                 | int          | NO   |     | 0       |                |
| complexity_level          | varchar(3)   | NO   |     | -1      |                |
| auto_update_pwd_frequency | tinyint      | NO   |     | 0       |                |
| auto_update_pwd_next_date | varchar(100) | NO   |     | 0       |                |
| encryption_type           | varchar(20)  | NO   |     | not_set |                |
| fa_icon                   | varchar(100) | YES  |     | NULL    |                |
+---------------------------+--------------+------+-----+---------+----------------+
21 rows in set (0,00 sec)
```

**Comment by whosht** [comment]
@nilsteampassnet: I have this problem after you fixed #3712. I commented it there (I cannot re-open the issue and as there is this issue I will write here).
@nilsteampassnet: Thx a lot for the fix. A few things I observed with the fix:
1. Use the long URL above in the URL field and saving the entry results in a wrong message
1.1. Observed: "Data is not valid - Field PW exceeds maximum length of"
1.2. Expected: "Data is not valid - Field URL exceeds maximum length of SOMENUMBER"
2. Use the long URL above in the URL field, save entry, get error, remove URL field content or put anything else in
2.1. Observed: Error Message appears
2.2. Entry should be saved
3. Edit an existing entry that has a description and save it.
3.1. Observed: Error Message appears regarding field "Description"
3.2. Entry should be saved
